### ISSUE TITLE
Display attribution

### ISSUE THREAD

**Issue by tchavet** [firstfloorsoftware/giphy_picker#17]
Hi,
Giphy will require starting on december 1st that all gifs assign proper attribution to the author. Is there an update planned for this plugin to display it in the GiphyImage class or do we have implement it ourselves by making a request to the api to get the author's name?
Thanks

**Comment by kozw** [comment]
Thanks for the heads-up, will look into it. It's probably best that giphy_picker addresses this.

Question: where do I find this requirement of proper author attribution per Dec 1st?

**Comment by tchavet** [comment]
Here is the email I received from them explaining everything https://mailchi.mp/0762226d308c/important-updates-for-your-giphy-integration

**Comment by aidangill-projects** [comment]
Yes this is important if you want to use the plugin in a production environment, otherwise you're limited to their development API only which only allows a small number of GIFs for testing purposes.

They also require their logo to be displayed saying 'Powered by GIPHY' - say at the bottom of the search screen.

From their terms of service:

https://support.giphy.com/hc/en-us/articles/360028134111-GIPHY-API-Terms-of-Service-

'You must label your application with “Powered by Giphy” and the Giphy logo..'

and

'You agree to properly attribute all pieces of content that you obtain through the API and at no time will you falsely attribute any such content. For the avoidance of doubt, this means that you won’t display any content in your applications without Giphy user and/or source attribution where available.'

I'm keen to use this plugin in a production environment so if it could be added in giphy_picker natively that would be awesome.

To obtain a Production API with proper usage limits you need to upload a video of the app with the above attributions etc.

**Comment by kozw** [comment]
A GiphyOverlay widget is already included which renders a 'Powered by GIPHY' overlay image over its content. This widget is used when previewing gif selections and can be used in your application as well. This should cover the Powered by Giphy requirement.

<img width="359" alt="Screenshot 2021-01-05 at 14 08 30" src="https://user-images.githubusercontent.com/10853103/103650152-e22cb400-4f5f-11eb-94ae-e05ec799b1a4.png">


I'm not sure how to implement the user/source attribution.

**Comment by aidangill-projects** [comment]
Ah I see, yes you're right thanks! For source attribution I'll implement this independently from the widget for the moment 👌

**Comment by aidangill-projects** [comment]
If it helps anyone else I've gotten production approval from GIPHY with the following changes:

1) Adding the powered by GIPHY logo on the search and preview screens
2) Putting a powered by GIPHY and a clickable attribution link next to the GIF when shown (if it exists), which I save in the database encoded along with URL, which is all provided by the gif object, no API interaction required

This may be overkill but the guidelines are quite vague - this at least is enough to get a production API key.

Photos attached:

![giphy_1](https://user-images.githubusercontent.com/53994442/104602540-6a533d80-5673-11eb-8770-dbc158787a7f.png)
![giphy_2](https://user-images.githubusercontent.com/53994442/104602553-6cb59780-5673-11eb-8457-2d63aaf24a6d.png)
![giphy_3](https://user-images.githubusercontent.com/53994442/104602560-6e7f5b00-5673-11eb-8c00-37e7871bf80f.png)

My hacky changes can be found in my public repo:

https://github.com/aidangill-projects

**Comment by jabcodes23** [comment]
How do I access the GiphyOverlay widget? That widget does not exist for me.

**Comment by zaidkazi** [comment]
Any updates on this?

**Comment by Bolling88** [comment]
> If it helps anyone else I've gotten production approval from GIPHY with the following changes:
> 
> 1. Adding the powered by GIPHY logo on the search and preview screens
> 2. Putting a powered by GIPHY and a clickable attribution link next to the GIF when shown (if it exists), which I save in the database encoded along with URL, which is all provided by the gif object, no API interaction required
> 
> This may be overkill but the guidelines are quite vague - this at least is enough to get a production API key.
>
> My hacky changes can be found in my public repo:
> 
> https://github.com/aidangill-projects

Can't you add a PR for this? Essential stuff really.

**Comment by sohaibnoman** [comment]
<img src="https://user-images.githubusercontent.com/42720743/134784939-664bc7bf-ddfe-47d4-a909-543b32057422.png" alt="drawing" width="300"/>

This is how Signal does it, a solution similar to this one would work even you don't have a preview or appbar!

**Comment by sohaibnoman** [comment]
I have now added a attribution mark that can toggled on and off and should support dark mode to, Examples below go as follow: iOS -> Android -> iOS with the parameter attributionMarkDarkMode = true (This does not change the canvas color, the canvas colors follow your theme, it only changes the image)

<img src="https://user-images.githubusercontent.com/42720743/134804810-1b473a45-19a0-43c8-b023-ef1c8dcde08a.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/42720743/134804815-198af323-48bb-4419-b69a-ba7034655273.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/42720743/134804914-724cff8a-2427-440e-ac75-1e92080625a9.png" alt="drawing" width="300"/>

This can be user even if the search bar and preview are removed. I have gotten production approval with this solution and made a pull request. Or you can use the solition from my github
